src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CSharpTypesThatOwnDisposableFieldsShouldBeDisposable.cs

```diff
@@ -29,7 +29,8 @@ public CSharpDisposableFieldAnalyzer(Compilation compilation)
             {
                 if (node is AssignmentExpressionSyntax assignment)
                 {
-                    if (assignment.Right is ObjectCreationExpressionSyntax &&
+                    // ImplicitObjectCreationExpression	8659
+                    if (assignment.Right.RawKind is (int)SyntaxKind.ObjectCreationExpression or 8659 &&
```

Comment:
```suggestion
                    if (assignment.Right.IsKind(SyntaxKind.ObjectCreationExpression, SyntaxKindEx.ImplicitObjectCreationExpression) &&
```

Reply:
Adding `SyntaxKindEx` won't be straightforward. It requires MS.CA.CSharp which is not available for use in Lightup as the shared project is referenced by `MS.CA.NetAnalyzers`, not language-specific layers. Trying to move the reference down to CSharp didn't seem trivial.

Reply:
I don't understand. This file is a C#-specific file?

Reply:
@sharwell Yup this file is C#-specific. But where do I add `SyntaxKindEx`? Can't add it to `Lightup` because it's not C#-specific.

Reply:
You can create a new C#-specific lightup folder for this project, or create a new shared project Analyzer.CSharp.Utilities (similar to how we have Refactoring.CSharp.Utilities)